File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dd = {}
        for i, num in enumerate(nums):
            diffs = target - num
            if diffs in dd.keys():
                return [dd[diffs], i]
            dd[num]=i

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> m;
        int n = nums.size();
        for(int i = 0 ; i < n ; i++)
        {
            int temp = nums[i];
            if(m.find(temp) != m.end())
            {
                return {m[temp], i};
            }
            m[target - temp] = i;
        }
        return {};
    }
};